### PR TITLE
Render graph support of ray traced ambient occlusion

### PR DESCRIPTION
**Pull request by anisunity**
The PR is on top of this one: https://github.com/Unity-Technologies/Graphics/pull/1411
Please check only the last two commit.

Doing the Rendergraph implementation for RTAO and required denoisers. Enabling the render graph compatibiliity on RTAO tests.

Testing status
Render graph tests (DXR) that were enabled are green
The non-RG tests are also green (DXR)

### CODE REVIEW

**Comment by sebastienlagarde** [comment]
@RemyUnity yes and no. Automation for DXR don't cover denoising part (as there is no temporal accumulation), and we have discover that even if for DXR we have green test, Chap have found various issue in the previous PR that have been done for conversion. 
Anis is aware of it now, so should have a better local test coverage but we can't rely on automation only (for DXR)

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDDiffuseDenoiser.cs [hunk 1]
```diff
@@ -144,5 +146,46 @@ static public void DenoiseBuffer(CommandBuffer cmd, DiffuseDenoiserParameters dd
...
+            {
+                // Cannot run in async
+                builder.EnableAsyncCompute(false);
+
+                // Fetch all the resources
+                passData.parameters = tfParameters;
+                passData.depthStencilBuffer = builder.ReadTexture(depthPyramid);
+                passData.normalBuffer = builder.ReadTexture(normalBuffer);
+                passData.noisyBuffer = builder.ReadTexture(noisyBuffer);
+                passData.intermediateBuffer = builder.ReadTexture(builder.WriteTexture(renderGraph.CreateTexture(new TextureDesc(Vector2.one, true, true)
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDDiffuseDenoiser.cs:172]: intermediateBuffer seems to be only used in this pass so in this case you can use CreateTransientTexture and remove Read/WriteTexture. Transient resources are only available for this pass and have implicit read/write operations.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDTemporalFilter.RenderGraph.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            {
+                // Cannot run in async
+                builder.EnableAsyncCompute(false);
+
+                // Fetch all the resources
+                passData.parameters = tfParameters;
+                passData.depthStencilBuffer = builder.ReadTexture(depthPyramid);
+                passData.normalBuffer = builder.ReadTexture(normalBuffer);
+                passData.motionVectorBuffer = builder.ReadTexture(motionVectorBuffer);
+                passData.velocityBuffer = renderGraph.defaultResources.blackTextureXR;
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDTemporalFilter.RenderGraph.cs:36]: What's the difference between motionVector and Velocity buffers?
Do we always pass a black texture? In this case no need to put it in the pass data but we can access the default resources directly inside the pass code.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDTemporalFilter.RenderGraph.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                passData.historyDepthTexture = renderGraph.ImportTexture(hdCamera.GetCurrentFrameRT((int)HDCameraFrameHistoryType.Depth));
+                passData.historyNormalTexture = renderGraph.ImportTexture(hdCamera.GetCurrentFrameRT((int)HDCameraFrameHistoryType.Normal));
+                passData.noisyBuffer = builder.ReadTexture(noisyBuffer);
+                passData.validationBuffer = builder.ReadTexture(builder.WriteTexture(renderGraph.CreateTexture(new TextureDesc(Vector2.one, true, true)
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDTemporalFilter.RenderGraph.cs:40]: If only used in this pass, use CreateTransientTexture (as above)

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDTemporalFilter.RenderGraph.cs (hunk 2 above)
Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDTemporalFilter.RenderGraph.cs:36]: Velocity is the velocity of the intersected surface by the rays, motion vector and the SS velocity of the pixel that casted the ray. It is always black  in this function, but the shader code is shared

**Comment by anisunity** [comment]
review corrections done

**Comment by anisunity** [comment]
included in this PR https://github.com/Unity-Technologies/Graphics/pull/1610

**Pull request closed without merging**